### PR TITLE
Rails 4

### PR DESCRIPTION
**Pull request by Frost**
I updated everything, basically.

### CODE REVIEW

File: app/models/person.rb [hunk 1]
```diff
@@ -1,40 +1,39 @@
...
   end
 
   def name
-    '%s %s' % [first_name, last_name]
+    "%s %s" % [first_name, last_name]
   end
 
   def total_purchased_amount
-    purchases.unpaid.inject(0) { |sum, x| sum += x.total }.to_f
+    purchases.unpaid.reduce(0.0) { |a, e| a + e.total }
```
Comment (Koronen) [inline app/models/person.rb]: `purchases.unpaid.sum(:total)`?

File: spec/controllers/purchases_controller_spec.rb [hunk 2]
```diff
@@ -29,14 +28,15 @@ def mock_purchase(stubs = {})
...
   describe "GET new" do
     it "assigns a new purchase as @purchase" do
-      purchase = double("purchase",
-                        :items => double("array", build: []),
-                        :person= => nil,
-                        :person_id= => nil
+      purchase = spy("purchase",
+                     :items => double("array", build: []),
+                     :person= => nil,
+                     :person_id= => nil
```
Comment (Koronen) [inline spec/controllers/purchases_controller_spec.rb]: Do we need to specify `:person=` and `:person_id=` for this spy?

File: app/models/person.rb (hunk 1 above)
Comment (Frost) [inline app/models/person.rb]: It's not that easy. `Purchase#total` is an instance method that aggregates `PurchaseItem#amount`.

I will have a look at actually doing this in a better way though.

**Comment by Koronen** [comment]
Minor comments, otherwise LGTM! :+1: Merge at your discretion.

File: spec/controllers/purchases_controller_spec.rb (hunk 2 above)
Comment (Frost) [inline spec/controllers/purchases_controller_spec.rb]: No.

**Pull request merged**